signalfx-async-wrapper.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const sfxHelper = require('./signalfx-helper');
+
+let coldStart = true;
+
+class SignalFxWrapper {
+  constructor(
+    originalObj,
+    originalFn,
+    originalEvent,
+    originalContext,
```

Comment:
What happened to originalCallback, like in the original wrapper?

Reply:
Please do correct me if I'm wrong, but when you use async/await, you can directly return the response to AWS without needing to use callback and that's the benefit of using async/await. Here is one example.
```
// https://aws.amazon.com/blogs/compute/node-js-8-10-runtime-now-available-in-aws-lambda/
let AWS = require('aws-sdk');
let lambda = new AWS.Lambda();
let data;

exports.handler = async (event) => {
    try {
        data = await lambda.getAccountSettings().promise();
    }
    catch (err) {
        console.log(err);
        return err;
    }
    return data;
};
```

Reply:
If the original lambda function had a callback, do you know if the callback will still get called or is it ignored? If it's technically possible to have a callback, then we'd still need to support it from the wrapper.

Reply:
If I add a parameter for original callback, it does get a callback function from Lambda so it is possible to call the callback. So I just did a quick research on this.

- Callback function and `return` statement essentially do the exact same thing. In both way, you can provide the result.
- Also even if you call the callback function with the result and return the result after that, it shows the exact same result as when performing only one option.

Reply:
For some reason, I recommend to do `return` instead of calling callback function.
1. Explicitly returning a result seems to be more encouraged. On local dev environment, if I run the function using callback, it shows the result is `undefined`.
![Screen Shot 2019-03-19 at 1 48 31 PM](https://user-images.githubusercontent.com/47250481/54629786-aead2c80-4a4e-11e9-9405-046c2b6e1ae8.png)
![Uploading Screen Shot 2019-03-19 at 1.49.07 PM.png…]()
2. It wouldn't be a good practice to use callback when we use async/await pattern.

Reply:
I realize that it defeats the purpose of using async when a callback is passed in. But if the user chooses to have a callback anyway in their original function, we don't want to just ignore it if their function would have used the callback when not using our wrapper.

Reply:
Just made a change to support callback. Now the user can use either a callback or return to conclude the action.